java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/IAtomicReferenceTest.java

```diff
@@ -53,22 +53,47 @@ String createString(int kb) {
     }
 
     @TimeStep(prob = 1)
-    public void write(ThreadState state) {
+    public void set(ThreadState state) {
         atomicReference.set(atomicReference.getName());
-        state.writes++;
+        state.ops++;
+    }
+
+    @TimeStep(prob = 0)
+    public void alter(ThreadState state) {
+        atomicReference.alter(state.identity);
+        state.ops++;
+    }
+
+    @TimeStep(prob = 0)
+    public void cas(ThreadState state) {
+        atomicReference.compareAndSet(v, v);
+        state.ops++;
+    }
+
+    @TimeStep(prob = 0)
+    public void casOptimisticConcurrencyControl(ThreadState state) {
+        String observed;
+        String newValue;
+        do {
```

Comment:
Since the value is never changed, this loop will always succeed. Not sure if that is very helpful since there is little contention this way.

Reply:
Yes, it's there to simply model the pattern. It will never loop as the cas will always succeed. Removing the loop obfuscates the pattern it models -- so, it's effectively documentation of the pattern.